kube/pod.go

```diff
@@ -218,7 +218,9 @@ func getEnvVars(role *model.Role, defaults map[string]string, secrets SecretRefM
 		}
 
 		if config.Secret {
-			secretKeyRef := helm.NewMapping("key", secrets[config.Name].Key, "name", secrets[config.Name].Secret)
```

Comment:
You need to wrap the new code into a `	if settings.CreateHelmChart { }` block and keep the old code in the `else` clause because the plain kube spec files don't have helm template expansion.

Also all the test changes need to be reverted.
`

Reply:
Ok. Oops.

Reply:
Handled with commit f670431aa64e514ac17c59006f764cef44442fce